### ISSUE TITLE
Add tests to `decompose_contrasts`

### ISSUE THREAD

**Issue by tsostarics** [tsostarics/contrastable#1]
Sonderegger 2021 notes that it can be useful to take the columns that utilize the contrast coded values from the model matrix and use them as predictors in the model directly. For example, you might have an ordered factor column `continuumStep` with 7 levels, which is coded with orthogonal polynomial contrasts. You can model your response in terms of the linear and quadratic components for the sake of parsimony if you want, but removing the higher order contrasts from the matrix requires you to do so using the contrasts argument in model fitting functions, which is not always available (eg in `brms::brm`). So, you can use `model.matrix` to extract the numeric values of the linear and quadratic components into new columns `lin` and `quad` and adding just these two terms as predictors in your model. This is somewhat more reliable and generalizable, but feels a bit hackish and could be done with a helper. Something mutate-friendly would be nice.

**Comment by tsostarics** [comment]
Revisiting this now, I've decided against using nonstandard evaluation to allow unquoted names to be used in favor of focusing more on vectorizing the functionality. Moreover, I've decided not to write this such that it's used within mutate. Instead I've written it to work more like `set_contrasts` such that it's its own step in a pipeline. This makes it a bit easier to handle passing arguments around and makes it more straightforward to add multiple columns to a dataframe. Commit d49b4ca provides the new implementation.

Tests need to be added for this though
